File: c/exo_12.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

int main(void) {
  int i, x;
  char year[5] = {0};
  time_t seconds = time(NULL);
  struct tm* current_time = localtime(&seconds);
  int date = current_time->tm_year + 1900;
  printf("En quelle année es-tu né(e) ?\n> ");
  fgets(year, 5, stdin);
  year[strcspn(year, "\n")] = 0;
  int y = date - atoi(year);
  for (i = atoi(year); i <= date; ++i){
      if(x == y)
	  printf("Il y a %d ans, tu avais la moitié de l'âge que tu as aujourd'hui\n", x);
      else
	  printf("Il y a %d ans, tu avais %d ans\n", y, x);
        x = x + 1;
        y = y - 1;
  }
}

File: rs/src/bin/exo_12.rs
use chrono::prelude::*;
use std::io;

fn main() {
    println!("En quelle année es-tu né(e) ?");
    let mut birth_buf = String::new();

    io::stdin().read_line(&mut birth_buf).unwrap();

    let current_year = Utc::today().year();
    let birth_year = birth_buf.trim().parse::<i32>().unwrap();

    if birth_year >= current_year {
        panic!("Tu ne peux pas être aussi jeune ?")
    }

    for i in birth_year..=current_year {
        let delta = current_year - i;
        let age = i - birth_year;
        if (age == delta) {
            println!(
                "Il y a {} ans, tu avais la moitié de l'age que tu as aujourd'hui.",
                delta
            )
        } else {
            println!("Il y a {} ans, tu avais {} ans.", delta, age)
        }
    }
}
